aait/backend/group-2B/Application.Tests/Features/Authentication/Handlers/Commands/RegisterUserCommandHandlerTest.cs

```diff
@@ -0,0 +1,133 @@
+        var jwtSettings = new JwtSettings
+        {
+            Secret = "Social-Sync-JWT-Secret-and-then-some",
+            Issuer = "SocialSync",
+            Audience = "SocialSync",
+            ExpiryMinutes = 60
+        };
+        var jwtOptions = Options.Create(jwtSettings);
+        _mapper = mapperConfig.CreateMapper();
+        _mockUnitOfWork = MockUnitOfWork.GetMockUnitOfWork();
+        _jwtGenerator = new JwtGenerator(new DateTimeProvider(), jwtOptions);
+        _passwordHasher = new PasswordHasher();
+        _handler = new RegisterUserCommandHandler(
+            _mockUnitOfWork.Object,
+            _jwtGenerator,
+            _mapper,
+            _passwordHasher
+        );
+    }
+
+    [Fact]
+    public async Task Valid_RegisterNewUser()
+    {
+        var newUser = new RegisterUserDto
+        {
+            FirstName = "User4",
+            LastName = "User4",
+            Email = "User4@gmail.com",
+            Username = "User4",
+            Password = "User4password",
+            Phone = "1234567890",
+        };
+
+        var result = await _handler.Handle(
+            new RegisterUserCommand { RegisterUserDto = newUser },
+            CancellationToken.None
+        );
+
+        result.ShouldBeOfType<CommonResponse<LoggedInUserDto>>();
+        result.IsSuccess.ShouldBeTrue();
+        result.Value.ShouldNotBeNull();
+
+        var Users = await _mockUnitOfWork.Object.UserRepository.GetAsync();
+        Users.Count.ShouldBe(4);
+    }
+
+    [Fact]
+    public async Task Invalid_EmailRegisterNewUser()
+    {
+        var newUser = new RegisterUserDto
+        {
+            FirstName = "User4",
+            LastName = "User4",
+            Email = "User4", // Not an email
+            Username = "User4",
+            Password = "User4password",
+            Phone = "1234567890",
+        };
+
+        var result = await _handler.Handle(
+            new RegisterUserCommand { RegisterUserDto = newUser },
+            CancellationToken.None
+        );
+
+        result.ShouldBeOfType<CommonResponse<LoggedInUserDto>>();
+        result.IsSuccess.ShouldBeFalse();
+        result.Value.ShouldBeNull();    }
+
+    [Fact]
+    public async Task Invalid_DuplicateEmailRegisterNewUsew()
+    {
+        var newUser = new RegisterUserDto
+        {
+            FirstName = "User4",
+            LastName = "User4",
+            Email = "User4@gmail.com",
+            Username = "User4",
+            Password = "User4password",
+            Phone = "1234567890",
+        };
+
+        var result = await _handler.Handle(
+            new RegisterUserCommand { RegisterUserDto = newUser },
+            CancellationToken.None
+        );
+
+        newUser = new RegisterUserDto
+        {
+            FirstName = "User4",
+            LastName = "User4",
+            Email = "User4@gmail.com",
+            Username = "User5",
+            Password = "User4password",
+            Phone = "1234567890",
+        };
```

Comment:
since you're not using it, remove the code snippet below:
newUser = new RegisterUserDto
        {
            FirstName = "User4",
            LastName = "User4",
            Email = "User4@gmail.com",
            Username = "User5",
            Password = "User4password",
            Phone = "1234567890",
        };

Reply:
I am trying to duplicate the email while using a new username and make it fail. The issue is I need to move it up before the result. Noted.